### PR TITLE
add new workflows + action

### PR DESCRIPTION
**Pull request by matanshk**
This PR follows the guidelines as described in the [documentation](https://github.com/kubescape/kubescape/tree/master/.github/workflows#readme).

### CODE REVIEW

**Comment by dwertent** [comment]
We should first merge [this](https://github.com/kubescape/workflows/pull/1)

/cc @matanshk

File: .github/workflows/a-pr-scanner.yaml [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+       SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+    name: PR Scanner
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - uses: actions/setup-go@v3
```
Comment (dwertent) [inline .github/workflows/a-pr-scanner.yaml:29]: [Question]: Why do we need to install go?

File: .github/workflows/a-pr-scanner.yaml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        uses: peter-evans/create-or-update-comment@v2.1.0
+        with:
+          issue-number:  ${{ github.event.pull_request.number }}
+          body: |
+            Scan results:
+            - License scan: ${{ steps.licenses-scan.outcome }}
+            - Credentials scan: ${{ steps.credentials-scan.outcome }}
+            - Vulnerabilities scan: ${{ steps.vulnerabilities-scan.outcome }}
+          reactions: 'eyes'
+          # token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
```
Comment (dwertent) [inline .github/workflows/a-pr-scanner.yaml:78]: [Request]: Remove the comment

Comment (dwertent) [reply .github/workflows/a-pr-scanner.yaml:78]: [Warninig]: This might break opening PRs from forks

File: .github/workflows/a-pr-scanner.yaml [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+        required: true
+        type: string
+      CLIENT:
+        description: 'Client name'
+        required: true
+        type: string
+
+
+jobs:
+  scanners:
```
Comment (dwertent) [inline .github/workflows/a-pr-scanner.yaml:17]: [TODO]: We should run this job from an external job

File: .github/workflows/01-code-review-approved.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+concurrency:
+  group: code-review-approved
+  cancel-in-progress: true
+
+jobs:
+
+  binary-build:
+    if: ${{ github.event.review.state == 'approved' && 
+            contains( github.event.pull_request.labels.*.name, 'trigger-integration-test') && 
+            github.event.pull_request.base.ref == 'master' }} ## run only if labeled as "trigger-integration-test" and base branch is master
```
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:18]: @vladklokun Notice here the `master` branch

File: .github/workflows/01-code-review-approved.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+concurrency:
+  group: code-review-approved
+  cancel-in-progress: true
+
+jobs:
+
+  binary-build:
+    if: ${{ github.event.review.state == 'approved' && 
+            contains( github.event.pull_request.labels.*.name, 'trigger-integration-test') &&
```
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:17]: [TODO]: @matanshk add labels to KS repo

File: .github/workflows/b-binary-build-and-e2e-tests.yaml [hunk 6]
```diff
@@ -0,0 +1,249 @@
+name: b-binary-build-and-e2e-tests
+on:
+  workflow_call:
+    inputs:
+      COMPONENT_NAME:
+        required: true
+        type: string
+        default: kubescape
```
Comment (dwertent) [inline .github/workflows/b-binary-build-and-e2e-tests.yaml:8]: [TODO]: Remove default if required

File: .github/workflows/b-binary-build-and-e2e-tests.yaml [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+          path: build/${{ matrix.os }}/
+          if-no-files-found: error
+
+  run-tests:
+    strategy:
+      fail-fast: false    
+      matrix:
+        TEST: ${{ fromJson(needs.binary-build.outputs.TEST_NAMES) }}
+    needs: binary-build
+    runs-on: ubuntu-20.04
```
Comment (dwertent) [inline .github/workflows/b-binary-build-and-e2e-tests.yaml:179]: [Comment]: This cannot change

File: .github/workflows/01-code-review-approved.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  cancel-in-progress: true
+
+jobs:
+
+  binary-build:
+    if: ${{ github.event.review.state == 'approved' && 
+            contains( github.event.pull_request.labels.*.name, 'trigger-integration-test') && 
+            github.event.pull_request.base.ref == 'master' }} ## run only if labeled as "trigger-integration-test" and base branch is master
+
+    uses: ./.github/workflows/b-binary-build-and-e2e-tests.yaml
```
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:20]: We should trigger this only if all secrets are found.

File: .github/workflows/02-release.yaml [hunk 9]
```diff
@@ -0,0 +1,55 @@
+name: 02-create_release
+
+on:
+  push:
+    tags:
+    - 'v*.*.*-rc.*'
+    # - 'v*.*.*'
```
Comment (dwertent) [inline .github/workflows/02-release.yaml:7]: [TODO]: Remove comment

File: .github/workflows/01-code-review-approved.yaml [hunk 10]
```diff
@@ -0,0 +1,44 @@
+name: 01-code_review_approved
+on:
+  pull_request_review:
+    types: [submitted]
+    branches:
+      - 'master'
+      - 'main'
```
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:7]: Add: `paths-ignore:`

File: .github/workflows/01-code-review-approved.yaml [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+
+  merge-to-master:
+    needs: binary-build
+    if: ${{ (always() && (contains(needs.*.result, 'success') || contains(needs.*.result, 'skipped')) && !(contains(needs.*.result, 'failure')) && !(contains(needs.*.result, 'cancelled'))) }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: merge-to-master
+        uses: pascalgn/automerge-action@v0.15.5
+        env:
+          GITHUB_TOKEN: "${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}"
```
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:39]: [Required] Check that secret is found

**Review by dwertent — CHANGES_REQUESTED** [review]

File: .github/workflows/b-binary-build-and-e2e-tests.yaml (hunk 7 above)
Comment (matanshk) [reply .github/workflows/b-binary-build-and-e2e-tests.yaml:179]: From the workflow perspective, it's can change without breaking nothing (if the systests successfully passed on Ubuntu 22)

File: .github/workflows/01-code-review-approved.yaml [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+      GO111MODULE: ""
+      GO_VERSION: "1.19"
+      RELEASE: ${{ github.ref_name}}
+      CLIENT: test
+    secrets: inherit
+
+
+  merge-to-master:
+    needs: binary-build
+    if: ${{ env.GH_PERSONAL_ACCESS_TOKEN &&
```
Comment (dwertent) [inline .github/workflows/01-code-review-approved.yaml:51]: [TODO]: We should have a step named `run-merge-to-master`, that step should store the output of the `if` statement so this step will look cleaner

File: .github/workflows/02-release.yaml [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    needs: [retag]
+    env:
+      CUSTOMER: ${{ secrets.CUSTOMER }}
+      USERNAME: ${{ secrets.USERNAME }}
+      PASSWORD: ${{ secrets.PASSWORD }}
+      CLIENT_ID: ${{ secrets.CLIENT_ID_PROD }}
+      SECRET_KEY: ${{ secrets.SECRET_KEY_PROD }}
+      REGISTRY_USERNAME: ${{ secrets.REGISTRY_USERNAME }}
+      REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+    if: ${{ env.CUSTOMER &&
```
Comment (dwertent) [inline .github/workflows/02-release.yaml:31]: [TODO]: We should have a step named `run-binary-build`, that step should store the output of the `if` statement so this step will look cleaner

**Comment by dwertent** [comment]
Please fix the CDO issue :)

File: .github/workflows/01-code-review-approved.yaml (hunk 10 above)
Comment (dwertent) [reply .github/workflows/01-code-review-approved.yaml:7]: @matanshk

**Comment by dwertent** [comment]
@matanshk I'm waiting for the github actions tests to pass before we merge.

@matthyx Notice, here we are overriding the changes you made regarding krew #1022 , we need to make sure to merge the two so we don't lose anything.

**Comment by matthyx** [comment]
> @matanshk I'm waiting for the github actions tests to pass before we merge.
> 
> @matthyx Notice, here we are overriding the changes you made regarding krew #1022 , we need to make sure to merge the two so we don't lose anything.

do you want me to take care of the rebase ?

**Comment by matanshk** [comment]
@dwertent 
All fixed from my side

**Review by dwertent — COMMENTED** [review]
What about the WF we have running now? Are you not deleting the files?

**Review by dwertent — APPROVED** [review]

**Pull request closed without merging**